code/go/0chain.net/smartcontract/dbs/event/authorizer_aggregate_test.go

```diff
@@ -0,0 +1,324 @@
+package event
+
+import (
+	"testing"
+
+	"0chain.net/chaincore/config"
+	faker "github.com/go-faker/faker/v4"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+func TestAuthorizerAggregateAndSnapshot(t *testing.T) {
+	t.Run("should update aggregates and snapshots correctly when a authorizer is added, updated or deleted", func(t *testing.T) {
+		// PartitionKeepCount = 10
+		// PartitionChangePeriod = 100
+		// For round 0 => authorizer_aggregate_0 is created for round from 0 to 100
+		const updateRound = int64(15)
+
+		eventDb, clean := GetTestEventDB(t)
+		defer clean()
+		eventDb.settings.Update(map[string]string{
+			"server_chain.dbs.settings.aggregate_period":        "10",
+			"server_chain.dbs.settings.partition_change_period": "100",
+			"server_chain.dbs.settings.partition_keep_count":    "10",
+		})
+
+		var (
+			expectedBucketId	int64
+			initialSnapshot		= Snapshot{ Round: 5 }
+			authorizerIds		= createAuthorizers(t, eventDb, 5, expectedBucketId)
+			authorizersBefore	[]Authorizer
+			authorizersAfter	[]Authorizer
+			authorizerSnapshots	[]AuthorizerSnapshot
+			expectedAggregates	[]AuthorizerAggregate
+			expectedSnapshots	[]AuthorizerSnapshot
+			err                 error
+		)
+		expectedBucketId = 5 % config.Configuration().ChainConfig.DbSettings().AggregatePeriod
+		err = eventDb.Store.Get().Model(&Snapshot{}).Create(&initialSnapshot).Error
+		require.NoError(t, err)
+
+		// Initial authorizers table image + force bucket_id for authorizers in bucket
+		err = eventDb.Get().Model(&Authorizer{}).Where("id IN ?", authorizerIds).Find(&authorizersBefore).Error
+		require.NoError(t, err)
+		authorizersInBucket := []string{ authorizersBefore[0].ID, authorizersBefore[1].ID, authorizersBefore[2].ID }
+		err = eventDb.Store.Get().Model(&Authorizer{}).Where("id IN ?", authorizersInBucket).Update("bucket_id", expectedBucketId).Error
+		require.NoError(t, err)
+		err = eventDb.Store.Get().Model(&Blobber{}).Where("id NOT IN ?", authorizersInBucket).Update("bucket_id", expectedBucketId + 1).Error
+		require.NoError(t, err)
+		err = eventDb.Get().Model(&Authorizer{}).Where("id IN ?", authorizerIds).Find(&authorizersBefore).Error
+		require.NoError(t, err)
+		err = eventDb.Get().Model(&AuthorizerSnapshot{}).Find(&authorizerSnapshots).Error
+		require.NoError(t, err)
+
+		expectedAggregates, expectedSnapshots = calculateAuthorizerAggregatesAndSnapshots(5, expectedBucketId, authorizersBefore, authorizerSnapshots)
+
+		// Initial run. Should register snapshots and aggregates of authorizers in bucket
+		eventDb.updateAuthorizerAggregate(5, 10, &initialSnapshot)
+		eventDb.Store.Get().Exec("DROP TABLE IF EXISTS authorizer_temp_ids")
```

Comment:
don't need it, table is unique for each transaction

Reply:
I need it bcz I do two successive calls to the function that creates it in the same test. If I deleted it then the second call will no create the table and will use the one from the prev run, which has some providers that I deleted and doesn't contain ones that I added.